libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1001,9 +1001,14 @@ void IRGeneratorForStatements::endVisit(IndexAccess const& _indexAccess)
 			}
 		});
 	}
-	else if (baseType.category() == Type::Category::Array)
+	else if (baseType.category() == Type::Category::Array || baseType.category() == Type::Category::ArraySlice)
 	{
-		ArrayType const& arrayType = dynamic_cast<ArrayType const&>(baseType);
+		// ArrayType const& arrayType = dynamic_cast<ArrayType const&>(baseType);
+		ArrayType const& arrayType =
+			baseType.category() == Type::Category::Array ?
+			dynamic_cast<ArrayType const&>(baseType) :
+			dynamic_cast<ArraySliceType const&>(baseType).arrayType();
+
```

Comment:
Maybe we should be a bit more conservative with assertions here as well (so far it is unclear how memory and storage slices will work, once we allow them):
```
if (baseType.category() == Type::Category::ArraySlice)
  solAssert(arrayType.dataStoredIn(DataLocation::Calldata) && arrayType.isDynamicallySized());
```

``isDynamicallySized()`` is probably superfluous, but we can still just remove it, when we allow slices for statically sized calldata arrays.

Reply:
I agree with asserting more